optimade/adapters/references/__init__.py

```diff
@@ -0,0 +1,17 @@
+from typing import Union
+
+from optimade.models import ReferenceResource
+
+from optimade.adapters.base import EntryAdapter
+
+
+__all__ = ("Reference",)
+
+
+class Reference(EntryAdapter):
+    """Lazy structure converter
+    :param structure: JSON OPTIMADE single structures resource entry.
+    """
+
+    ENTRY_RESOURCE = ReferenceResource
+    _type_converters = {}
```

Comment:
I was thinking of this, but didn't want to throw myself into researching bibjson.
Instead, I was hoping someone with the knowledge of it would add a converter, either to this PR or afterwards - and to any other type of desired conversion format.
However, I can give a short look into it now, and if it takes too long, I'll make an issue for it once this is merged or ping you if you want to give it a go before.

Reply:
I think this ties in with the comment above about a standard `json` "converter"